posthog/api/plugin.py

```diff
@@ -0,0 +1,147 @@
+            raise ValidationError('Plugin with name "{}" already installed!'.format(validated_data["name"]))
+        validated_data["archive"] = download_plugin_github_zip(validated_data["url"], validated_data["tag"])
+        validated_data["from_web"] = True
+        plugin = super().create(validated_data)
+        reload_plugins_on_workers()
+        return plugin
+
+    def update(self, plugin: Plugin, validated_data: Dict, *args: Any, **kwargs: Any) -> Plugin:  # type: ignore
+        validated_data["archive"] = download_plugin_github_zip(plugin.url, plugin.tag)
+        response = super().update(plugin, validated_data)
+        reload_plugins_on_workers()
+        return response
+
+
+class PluginViewSet(viewsets.ModelViewSet):
+    queryset = Plugin.objects.all()
+    serializer_class = PluginSerializer
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        if self.action == "get" or self.action == "list":  # type: ignore
+            if can_install_plugins_via_api() or can_configure_plugins_via_api():
+                return queryset
+        else:
+            if can_install_plugins_via_api():
+                # block update/delete for plugins that come from posthog.json
+                return queryset.filter(from_json=False)
+        return queryset.none()
+
+    @action(methods=["GET"], detail=False)
+    def repository(self, request: request.Request):
+        if not can_install_plugins_via_api():
+            raise ValidationError("Plugin installation via the web is disabled!")
+        url = "https://raw.githubusercontent.com/PostHog/plugins/main/repository.json"
+        plugins = requests.get(url)
+        return Response(json.loads(plugins.text))
+
+    @action(methods=["GET"], detail=False)
+    def status(self, request: request.Request):
+        if not can_install_plugins_via_api():
+            raise ValidationError("Plugin installation via the web is disabled!")
+
+        ping = get_client().get("@posthog-plugin-server/ping")
+        if ping:
+            ping_datetime = parser.isoparse(ping)
+            if ping_datetime > now() - relativedelta(seconds=30):
+                return Response({"status": "online"})
+
+        return Response({"status": "offline"})
+
+    def destroy(self, request: request.Request, *args, **kwargs) -> Response:
+        response = super().destroy(request, *args, **kwargs)
+        reload_plugins_on_workers()
+        return response
+
+
+class PluginConfigSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = PluginConfig
+        fields = ["id", "plugin", "enabled", "order", "config", "error"]
+        read_only_fields = ["id"]
+
+    def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> PluginConfig:
+        if not can_configure_plugins_via_api():
+            raise ValidationError("Plugin configuration via the web is disabled!")
+        request = self.context["request"]
+        validated_data["team"] = request.user.team
```

Comment:
As a new endpoint, this would be nicer if it used `drf-extensions`'s router nesting so that can make endpoints more flexible than by just using `request.user` everywhere (right now nesting is implemented on the `/api/organization` endpoint.

Reply:
I would like to revisit this in the next iteration when we add proper support for role-based permissions (e.g. the project admin can install plugins, regular members can't, etc). I hope it's OK to keep it as is for now.